rpc/src/main/scala/server/handlers/GrpcServerHandler.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package freestyle.rpc
+package server.handlers
+
+import cats.data.Kleisli
+import freestyle.rpc.server.{GrpcServer, GrpcServerOps}
+import freestyle.Capture
+import io.grpc.{Server, ServerServiceDefinition}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration.TimeUnit
+
+class GrpcServerHandler[F[_]](implicit C: Capture[F])
+    extends GrpcServer.Handler[GrpcServerOps[F, ?]] {
+
+  def start: GrpcServerOps[F, Server] = {
+
+    Runtime.getRuntime.addShutdownHook(new Thread() {
+      override def run(): Unit = {
+        shutdown
+        (): Unit
+      }
+    })
+
+    Kleisli(s => C.capture(s.start()))
+  }
+
+  def getPort: GrpcServerOps[F, Int] = Kleisli(s => C.capture(s.getPort))
```

Comment:
Maybe we can create a small helper method for this `Kleisli(s => C.capture(s.foo))` pattern?

Something like

```scala
private def captureWithServer[A](f: Server => A] : GrpcServerOps[F, A] = Kleisli(s => C.capture(f(s))`

def getPort: GrpcServerOps[F, Int] = captureWithServer(_.getPort)
```